iotilebuild/iotile/build/config/site_scons/trub_script.py

```diff
@@ -367,6 +368,30 @@ def combine_trub_scripts(trub_scripts_list, out_file):
         outfile.write(new_script.encode())
 
 
+def get_device_checksum(trub_scripts_list, checksum_address):
+    """Returns the device checksum from a trub file if it exists"""
+
+    resolver = ProductResolver.Create()
+
+    files = [resolver.find_unique("trub_script", x).full_path for x in trub_scripts_list]
+
+    for script in files:
+        if not os.path.isfile(script):
+            raise BuildError("Path to script is not a valid file.",
+                             script=script)
+
+        trub_binary = _get_trub_binary(script)
+        trub_script = UpdateScript.FromBinary(trub_binary)
+
+        for record in trub_script.records:
+            if isinstance(record, SetConfigRecord) and\
+            record.config_id == checksum_address:
+                device_checksum = record.data
+                return device_checksum
+
+    return None
+
+
```

Comment:
Searches through a list of TRUB scripts and looks for a record where the config id matches the checksum address.